Release fails when no files have been changed after running the build-task

Hi @RomanHotsiy,

I find your library very useful, but once in a while I am running into the case that I change some dependencies in my project, e.g. moving some npm dependencies from devDependencies to dependencies.

As this does not have any direct impact on the files, it will have an impact on how my library will be installed and therefore should be tagged as a new release, shouldn't it?

But as no files have changed, the branch-release task fails.
What would be the best way to fix this? Would love to hear your thoughts and would like to contribute a pull-request!

Thanks!
Stefan

Hi @RomanHotsiy,

it looks something like this:

`> branch-release -s build

set -e
exec git stash
No local changes to save
exec git show HEAD:package.json { silent: true }
exec git ls-remote -t origin { silent: true }
set +e
exec git fetch --unshallow
fatal: --unshallow on a complete repository does not make sense
set -e
rm -rf dist
exec git ls-remote -h origin { silent: true }
exec git fetch -f origin releases:releases
exec git checkout releases
Switched to branch 'releases'
exec git merge -X theirs @{-1}
Merge made by the 'recursive' strategy.
 package.json | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)
exec npm run-script build

> media-library-starter-template@1.0.1 build /Users/stefan/Documents/Development/media-library-starter-template
> n-clean ./dist/ && rollup -c rollup.config.js && rollup -c rollup.config.prod.js


sources/index.ts → ./dist/mediaLibrary.js...
created ./dist/mediaLibrary.js in 1.1s

sources/index.ts → ./dist/mediaLibrary.min.js...
created ./dist/mediaLibrary.min.js in 1.3s
exec git add dist -f
warning: CRLF will be replaced by LF in dist/index.d.ts.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in dist/mediaLibrary.js.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in dist/mediaLibraryModule.d.ts.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in dist/mediaLibraryService.d.ts.
The file will have its original line endings in your working directory.
exec git commit -m "Release v1.0.1"
On branch releases
nothing to commit, working tree clean
exec: internal error
exec git reset --hard
HEAD is now at c072282 Merge branch 'master' into releases
exec git checkout @{-1}
Your branch is up-to-date with 'origin/master'.
Switched to branch 'master'
exec git stash pop
No stash found.
/Users/stefan/Documents/Development/media-library-starter-template/node_modules/shelljs/src/common.js:417
      if (config.fatal) throw e;
                        ^

Error: exec: internal error
    at Object.error (/Users/stefan/Documents/Development/media-library-starter-template/node_modules/shelljs/src/common.js:128:27)
    at _exec (/Users/stefan/Documents/Development/media-library-starter-template/node_modules/shelljs/src/exec.js:292:12)
    at /Users/stefan/Documents/Development/media-library-starter-template/node_modules/shelljs/src/common.js:350:23
    at switchBack (/Users/stefan/Documents/Development/media-library-starter-template/node_modules/branch-release/index.js:50:3)
    at branchRelease (/Users/stefan/Documents/Development/media-library-starter-template/node_modules/branch-release/index.js:37:5)
    at main (/Users/stefan/Documents/Development/media-library-starter-template/node_modules/branch-release/index.js:20:3)
    at Object.<anonymous> (/Users/stefan/Documents/Development/media-library-starter-template/node_modules/branch-release/index.js:116:3)
    at Module._compile (internal/modules/cjs/loader.js:805:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:816:10)
    at Module.load (internal/modules/cjs/loader.js:672:32)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! media-library-starter-template@1.0.1 release: `branch-release -s build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the media-library-starter-template@1.0.1 release script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/stefan/.npm/_logs/2019-06-25T05_59_01_157Z-debug.log`

--- a/index.js
+++ b/index.js
@@ -80,7 +80,7 @@ function buildAndPublish(version) {
   exec(`npm run-script ${BUILD_SCRIPT}`);
   exec(`git add ${DIST_DIR} -f`);
   const commitMessage = COMMIT_MESSAGE.replace('%ver%', version);
-  exec(`git commit -m "${commitMessage}"`);
+  exec(`git commit -m "${commitMessage}" --allow-empty`);
   exec(`git tag v${version}`);
   let remote = 'origin';
   if (GH_TOKEN) {

